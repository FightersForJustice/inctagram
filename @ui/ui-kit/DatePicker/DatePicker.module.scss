@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');

.DatePicker {
  border-radius: 2px;
  border: 1px solid var(--dark-100, #4c4c4c);
  font-size: 16px;
  color: white;
  background-color: var(--black-a5);
  box-shadow: 0 0 0 1px var(--black-a9);
  padding: 6px 12px;
  line-height: 24px;

  &:hover {
    box-shadow: 0 0 0 1px black;
  }
  
  &:focus {
    box-shadow: 0 0 0 2px black;
  }
  
  &::selection {
    background-color: var(--black-a9);
    color: white;
  }

}


