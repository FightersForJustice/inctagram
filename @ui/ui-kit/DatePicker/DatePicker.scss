@use '../../design/settings/colors.scss' as colors;
@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');

.UIKitDatePicker {

  & .MuiInputBase-input {
    color: colors.$light-100;
    padding: 6px 12px;
    line-height: 24px;
    font-size: 16px;
    font-family: 'Inter';
    width: 110px;
    z-index: 10;
  }
  &--open {
    & .MuiInputBase-input {
      width: 250px;
    }
    & fieldset{
      padding-right: 288px;
      
    }
  }

  & .MuiOutlinedInput-notchedOutline {
    border-color: colors.$dark-300;
    border-radius: 2px;
    width: 158px;
  }

  & .css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root {
    &:hover .MuiOutlinedInput-notchedOutline {
      border-color: colors.$dark-100;
      background-color: colors.$dark-500;
    }

    &.Mui-focused .MuiOutlinedInput-notchedOutline {
      border-color: colors.$primary-700;
    }

    &.Mui-disabled .MuiOutlinedInput-notchedOutline {
      border-color: colors.$dark-300;
      background-color: colors.$dark-500;
    }

    &.Mui-error {
      .MuiOutlinedInput-notchedOutline {
        border-color: colors.$danger-500;
      }
      .MuiInputBase-input {
        color: colors.$danger-500;
      }
      .MuiButtonBase-root {
        filter: invert(18%) sepia(87%) saturate(3975%) hue-rotate(339deg) brightness(100%) contrast(98%);
      }
    }
  }

  & .css-nxo287-MuiInputBase-input-MuiOutlinedInput-input.Mui-disabled {
    -webkit-text-fill-color: colors.$light-900;
  }

  & .MuiButtonBase-root {
    position: absolute;
    right: 22px;
    padding: 0;
    filter: invert(1);
    z-index: 10;
  }

  & .MuiPickersLayout-root {
    padding: 20px;
  }
}

.MuiDateCalendar-root {
  width: 300px;
  height: 348px;
  background-color: colors.$dark-500;;
  color: colors.$light-100;

  .MuiTypography-caption {
    color: colors.$dark-100;;
  }

  .MuiPickersDay-root {
    color: colors.$light-100;;
    font-size: 16px;
    font-family: 'Inter';
    transition: .2s;

    &:hover,
    &:active {
      background-color: colors.$primary-900;
    }

    &:focus {
      background-color: inherit;
      border: 2px solid colors.$primary-300;
    }

  }

  .MuiPickersDay-today {
    color: colors.$primary-500;
    font-weight: 400;
    border: none;
  }

  .css-1jsy6pn-MuiButtonBase-root-MuiPickersDay-root:not(.Mui-selected) {
    border: none;
  }

  .Mui-selected {
    transition: all .2s;
    border: 4px solid colors.$primary-900;
    background-color: colors.$primary-500;
    color: colors.$light-100;

    &:focus {
      border: 4px solid colors.$primary-900;
      background-color: colors.$primary-500;
    }
  }

  .MuiSvgIcon-root {
    filter: invert(1);
  }

  .MuiPickersDay-dayOutsideMonth {
    color: colors.$dark-300;
  }

  .MuiPickersArrowSwitcher-button:hover,
  .MuiPickersCalendarHeader-switchViewButton:hover {
    background-color: colors.$dark-300;
  }

  .MuiDayCalendar {
    &-monthContainer {
      margin-top: 5px;
    }
  
    &-header {
      margin-top: 10px;
    }
  }

}