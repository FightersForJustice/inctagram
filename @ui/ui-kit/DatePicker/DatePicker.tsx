import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
import { DatePicker } from '@mui/x-date-pickers/DatePicker'
import './DatePicker.scss'
import 'dayjs/locale/ru'
import 'dayjs/locale/en'
import { useTranslation } from 'react-i18next'
import { Dispatch, useEffect, useState } from 'react'
import classNames from 'classnames'

type DatePickerTypes = {
  value?: Date
  setValue?: Dispatch<React.SetStateAction<any>>
}

export const saveToArray = (setValue: Dispatch<React.SetStateAction<any>>, name = 'date') => {
  return (value: Date) => {
    setValue((prevData: any) => ({
      ...prevData,
      [name]: value,
    }))
  }
}

export const MainDatePicker = ({ value, setValue }: DatePickerTypes) => {
  const { t } = useTranslation()
  const [date, setDate] = useState(null)
  const [pickerState, setPickerState] = useState('UIKitDatePicker--close')
  useEffect(() => {
    if (!setValue) return
    setValue = setDate
  }, [])
  return (
    <div className={classNames('UIKitDatePicker', pickerState)}>
      <LocalizationProvider dateAdapter={AdapterDayjs} adapterLocale={t('locale')}>
        <DatePicker
          onOpen={() => setPickerState('UIKitDatePicker--open')}
          onClose={() => setPickerState('UIKitDatePicker--close')}
          showDaysOutsideCurrentMonth
          dayOfWeekFormatter={(day) => day}
          onChange={(e: any) => setValue && setValue(e.$d)}
          value={value ? value : null}
        />
      </LocalizationProvider>
    </div>
  )
}

// export const TestComp = () => {
//   const [test, setTest] = useState<{date?: Date}>({})
//   return <div>
//     <MainDatePicker value={test.date} setValue={saveToArray(setTest, 'dateOfBirth')} />
//     <button onClick={() => console.log(test)}>test</button>
//   </div>
// }
