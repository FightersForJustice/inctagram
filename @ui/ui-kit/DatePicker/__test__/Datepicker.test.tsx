import { render, cleanup, screen, fireEvent } from '@testing-library/react'
import { MainDatePicker } from '../DatePicker'
describe('date picker renders correctly', () => {
  it('date picker render', () => {
    const { asFragment } = render(<MainDatePicker />)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('Date picker', () => {
  afterEach(cleanup)

  it('renders correctly', () => {
    render(<MainDatePicker />)

    expect(screen.getByPlaceholderText('MM/DD/YYYY')).toBeInTheDocument()
  })

  it('should be disabled when disabled prop is true', () => {
    render(<MainDatePicker disabled />)

    const datePickerElement = screen.getByPlaceholderText('MM/DD/YYYY')
    expect(datePickerElement).toBeDisabled()
  })

  it('should display error', () => {
    render(<MainDatePicker value="99/99/9999" />)

    const datePickerElement = screen.getByPlaceholderText('MM/DD/YYYY')
    expect(datePickerElement.parentElement).toHaveClass('Mui-error')
  })

  it('should display value', () => {
    render(<MainDatePicker value="12/12/2000" />)

    const datePickerElement = screen.getByPlaceholderText('MM/DD/YYYY')
    expect(datePickerElement).toHaveAttribute('value')
    expect(datePickerElement.getAttribute('value')).toBe('12/12/2000')
  })

  it('should display lower border value', () => {
    render(<MainDatePicker value="01/01/1900" />)

    const datePickerElement = screen.getByPlaceholderText('MM/DD/YYYY')

    expect(datePickerElement).toHaveAttribute('value')
    expect(datePickerElement.parentElement).not.toHaveClass('Mui-error')
    expect(datePickerElement.getAttribute('value')).toBe('01/01/1900')
  })

  it('should display lower out of border value with error', () => {
    render(<MainDatePicker value="12/31/1899" />)

    const datePickerElement = screen.getByPlaceholderText('MM/DD/YYYY')

    expect(datePickerElement).toHaveAttribute('value')
    expect(datePickerElement.parentElement).toHaveClass('Mui-error')
    expect(datePickerElement.getAttribute('value')).toBe('12/31/1899')
  })

  it('should display higher border value', () => {
    render(<MainDatePicker value="12/31/2099" />)

    const datePickerElement = screen.getByPlaceholderText('MM/DD/YYYY')

    expect(datePickerElement).toHaveAttribute('value')
    expect(datePickerElement.parentElement).not.toHaveClass('Mui-error')
    expect(datePickerElement.getAttribute('value')).toBe('12/31/2099')
  })

  it('should display higher out of border value with error', () => {
    render(<MainDatePicker value="01/01/2100" />)

    const datePickerElement = screen.getByPlaceholderText('MM/DD/YYYY')

    expect(datePickerElement).toHaveAttribute('value')
    expect(datePickerElement.parentElement).toHaveClass('Mui-error')
    expect(datePickerElement.getAttribute('value')).toBe('01/01/2100')
  })

  it('should open picker when calendar button clicked', () => {
    render(<MainDatePicker />)

    const datePickerButtonElement = screen.getByRole('button')
    fireEvent.click(datePickerButtonElement)
    expect(screen.getByRole('grid')).toBeInTheDocument()
  })

  it('should focus input when it`s clicked', () => {
    render(<MainDatePicker />)

    const datePickerElement = screen.getByPlaceholderText('MM/DD/YYYY')
    fireEvent.click(datePickerElement)
    expect(datePickerElement).toBeEnabled()
  })

  it('should focus input when label clicked', () => {
    render(
      <div>
        <label htmlFor="id" id="label">
          label
        </label>
        <MainDatePicker id="id" />
      </div>
    )

    const labelElement = screen.getByText('label')
    fireEvent.click(labelElement)
    expect(screen.getByPlaceholderText('MM/DD/YYYY')).toBeEnabled()
  })

  it('should change the date when "number" button clicked', () => {
    render(<MainDatePicker value="10/10/2000" />)

    fireEvent.click(screen.getByRole('button'))
    fireEvent.click(screen.getByRole('gridcell', { name: '24' }))
    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('10/24/2000')
  })

  it('should change the month when "next month" button clicked', () => {
    render(<MainDatePicker value="10/10/2000" />)

    fireEvent.click(screen.getByRole('button'))
    fireEvent.click(screen.getByTitle('Next month'))
    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('11/20/2000')
  })

  it('should change the month when "previous month" button clicked', () => {
    render(<MainDatePicker value="10/10/2000" />)

    fireEvent.click(screen.getByRole('button'))
    fireEvent.click(screen.getByTitle('Previous month'))
    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('09/20/2000')
  })

  it('should change the year when "next month" button clicked on last month', () => {
    render(<MainDatePicker value="12/10/2000" />)

    fireEvent.click(screen.getByRole('button'))
    fireEvent.click(screen.getByTitle('Next month'))
    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('01/20/2001')
  })

  it('should change the year when "previous month" button clicked on first month', () => {
    render(<MainDatePicker value="01/10/2000" />)

    fireEvent.click(screen.getByRole('button'))
    fireEvent.click(screen.getByTitle('Previous month'))
    fireEvent.click(screen.getAllByRole('gridcell', { name: '20' })[0])
    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('12/20/1999')
  })

  it('should disable "Previous month" button on first month of first year', () => {
    render(<MainDatePicker value="01/01/1900" />)

    fireEvent.click(screen.getByRole('button'))
    expect(screen.getByTitle('Previous month')).toBeDisabled()
  })

  it('should disable "Next month" button on last month of last year', () => {
    render(<MainDatePicker value="12/31/2099" />)

    fireEvent.click(screen.getByRole('button'))
    expect(screen.getByTitle('Next month')).toBeDisabled()
  })

  it('should open year selector when "year" arrow button clicked', () => {
    render(<MainDatePicker value="01/10/2000" />)

    fireEvent.click(screen.getByRole('button'))
    fireEvent.click(screen.getByLabelText('calendar view is open, switch to year view'))
    expect(screen.getByText('1900')).toBeInTheDocument()
    expect(screen.getByText('2099')).toBeInTheDocument()
  })

  it('should change year when "year" button clicked', () => {
    render(<MainDatePicker value="10/10/2000" />)

    fireEvent.click(screen.getByRole('button'))
    fireEvent.click(screen.getByLabelText('calendar view is open, switch to year view'))
    fireEvent.click(screen.getByText('1950'))
    expect(screen.getByPlaceholderText('MM/DD/YYYY').getAttribute('value')).toBe('10/10/1950')
  })
  it('should focus number button', () => {
    render(<MainDatePicker value="10/10/2000" />)

    fireEvent.click(screen.getByRole('button'))
    const dateNumberButton = screen.getByRole('gridcell', { name: '24' })
    fireEvent.focus(dateNumberButton)
    expect(dateNumberButton).toHaveClass('Mui-focusVisible')
  })

  it('should have "selected" class on selected number button', () => {
    render(<MainDatePicker value="10/10/2000" />)

    fireEvent.click(screen.getByRole('button'))
    const dateNumberButton = screen.getByRole('gridcell', { name: '10' })
    expect(dateNumberButton).toHaveClass('Mui-selected')
  })
})
