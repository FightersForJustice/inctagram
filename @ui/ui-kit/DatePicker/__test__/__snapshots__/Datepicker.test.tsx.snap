// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date picker renders correctly should render default correctly 1`] = `
<DocumentFragment>
  <div
    class="UIKitDatePicker UIKitDatePicker--close"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          inputmode="text"
          placeholder="MM/DD/YYYY"
          type="text"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            aria-label="Choose date"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CalendarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`date picker renders correctly should render default with open picker correctly 1`] = `
<DocumentFragment>
  <div
    class="UIKitDatePicker UIKitDatePicker--open"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          inputmode="tel"
          placeholder="MM/DD/YYYY"
          type="text"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            aria-label="Choose date"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CalendarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`date picker renders correctly should render disabled correctly 1`] = `
<DocumentFragment>
  <div
    class="UIKitDatePicker UIKitDatePicker--close"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          disabled=""
          id=""
          inputmode="text"
          placeholder="MM/DD/YYYY"
          type="text"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            aria-label="Choose date"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CalendarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
              />
            </svg>
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`date picker renders correctly should render error correctly 1`] = `
<DocumentFragment>
  <div
    class="UIKitDatePicker UIKitDatePicker--close"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="true"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          inputmode="text"
          placeholder="MM/DD/YYYY"
          type="text"
          value="MM/DD/YYYY"
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            aria-label="Choose date"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CalendarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
      >
        Error!
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`date picker renders correctly should render error with open picker correctly 1`] = `
<DocumentFragment>
  <div
    class="UIKitDatePicker UIKitDatePicker--open"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="true"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id=""
          inputmode="tel"
          placeholder="MM/DD/YYYY"
          type="text"
          value="MM/DD/YYYY"
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            aria-label="Choose date"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CalendarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-focused MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
      >
        Error!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
